### PR TITLE
pkg/datapath/loader: log versions

### PR DESCRIPTION
**Pull request by florianl**
Signed-off-by: Lehner Florian <dev@der-flo.net>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](http://docs.cilium.io/en/stable/contributing/contributing/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
 `This PR does not add or remove functionality.`
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue. 
  `There exists no particular GitHub issue`
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](http://docs.cilium.io/en/stable/contributing/contributing/#dev-coo).
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

This PR add the logging of the versions of used tools.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10096)

### CODE REVIEW

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

File: pkg/datapath/loader/compile.go [hunk 1]
```diff
@@ -257,9 +257,23 @@ func compile(ctx context.Context, prog *progInfo, dir *directoryInfo, debug bool
...
+	}
+	versionCmd = exec.CommandContext(ctx, linker, "--version")
+	linkerVersion, err := versionCmd.CombinedOutput(scopedLog, debug)
+	if err != nil {
+		return err
+	}
+	scopedLog.WithFields(logrus.Fields{
+		compiler: string(compilerVersion),
+		linker:   string(linkerVersion),
+	}).Debug("JoinEP: Tool versions")
```
Comment (joestringer) [inline pkg/datapath/loader/compile.go]: The CI looks for `JoinEP: ` as a failure string, which is why the CI failed. Suggestion below:

```suggestion
	}).Debug("Compiling datapath")
```

**Review by joestringer — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by florianl** [comment]
The failing ginkgo-test reports:
```[Pipeline] sh
15:52:54  + /usr/local/bin/cleanup
15:52:54  Error response from daemon: conflict: unable to delete 620fc8f53b82 (must be forced) - image is referenced in multiple repositories
15:52:54  Error response from daemon: conflict: unable to delete 620fc8f53b82 (must be forced) - image is referenced in multiple repositories
15:52:54  Error response from daemon: conflict: unable to delete 1f2e19f10224 (must be forced) - image is referenced in multiple repositories
```
I guess, this is a problem, I can not fix or is caused by this PR.

**Comment by joestringer** [comment]
I think it was just a temporary infrastructure issue:
```
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/cilium/cilium
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:894)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1161)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1192)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:124)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:144)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:67)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:293)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/cilium/cilium +refs/pull/*:refs/remotes/origin/pr/* --depth=1" returned status code 128:
stdout: 
stderr: fatal: expected shallow/unshallow, got ERR upload-pack: not our ref a3d3f0bced06e8d49dc377904044b31d90e3e8c1
fatal: The remote end hung up unexpectedly

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2042)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1761)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$400(CliGitAPIImpl.java:72)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:442)
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:892)
	... 8 more
ERROR: Error fetching remote repo 'origin'
Setting status of c37318141f68d11ab831c96bb1fd8695c991060d to FAILURE with url https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17263/ and message: 'Build finished. '
Using context: Cilium-Ginkgo-Tests
ERROR: Maximum checkout retry attempts reached, aborting
```
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17263/console

I'll kick the CI again.

**Comment by joestringer** [comment]
test-me-please

**Comment by florianl** [comment]
it looks like, the automated tests are not happy with this PR again:
```
21:08:00  + /usr/local/bin/cleanup
21:08:00  VBoxManage: error: Machine 'f2bad381-d637-44f3-8f4a-7d51c0499111' is not currently running
21:08:00  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
21:08:00  VBoxManage: error: Machine 'e30cee99-1a61-4b76-a2f9-8ea5eb1e3914' is not currently running
21:08:01  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
```

I'm sorry for the noise.

**Comment by joestringer** [comment]
Given the change occurring here, it seems unlikely that the most recent CI failures are related.

However, there are a bunch of test failures here and some of them are quite concerning (deadlock mentions, endpoints going into `not-ready` state, etc. I think we should file issues and follow up on the individual failures here to improve the stability of the tree. Assuming that we've saved all of these failures or followed up that they're already addressed in master, I'd be happy to kick off the CI again to test this PR.

**Comment by joestringer** [comment]
Filed #10274 #10273 #10278 , we can try again (but those issues are not yet resolved so the CI run may still fail)

**Comment by joestringer** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
@joestringer the PR has passed CI, are the changes good to you?

**Review by joestringer — APPROVED** [review]

**Pull request merged**